shell/components/nav/WindowManager/ContainerLogs.vue

```diff
@@ -273,11 +315,47 @@ export default {
         console.error('Connect Error', e); // eslint-disable-line no-console
       });
 
+      let logBuffer = [];
+      let truncatedLog = '';
+
       this.socket.addEventListener(EVENT_MESSAGE, (e) => {
-        const data = base64Decode(e.detail.data);
+        const b = base64DecodeToBuffer(e.detail.data.replace(/[-_]/g, (char) => char === '-' ? '+' : '/'));
+        const isTruncated = isLogTruncated(b);
+
+        if (isTruncated === true) {
+          logBuffer.push(...b);
+
+          return;
+        }
+
+        let d;
+
+        if (logBuffer.length > 0) {
+          d = ab2str(Uint8Array.of(...logBuffer, ...b));
+          logBuffer = [];
+        } else {
+          d = b.toString();
+        }
+        let data = d;
+
+        if (truncatedLog) {
+          data = `${ truncatedLog }${ d }`;
+          truncatedLog = '';
+        }
+
+        if (!d.endsWith('\n')) {
+          const lines = data.trim().split(/\n/);
 
+          if (lines.length === 1) {
+            truncatedLog = data;
+
+            return;
+          }
+          data = lines.slice(0, -1).join('\n');
+          truncatedLog = lines.slice(-1);
+        }
         // Websocket message may contain multiple lines - loop through each line, one by one
-        data.split('\n').forEach((line) => {
+        data.trim().split('\n').forEach((line) => {
```

Comment:
understand why you do the `trim` but `data.trim().split('\n')` I feel it's again a sequence of operations that might throw an error if something doesn't start off with the correct type or expected content